src/index.ts

```diff
@@ -29,6 +29,13 @@ export type Deferred<T> = Exclude<T, any[]>;
 export type PickDeferred<T, K extends keyof T> = {
 	[P in K]: Deferred<T[P]>;
 };
+export interface WebResource {
```

Comment:
@Page- Previous version of this was using a direct "{}" for the WebResource type. Although I see that it is a bit odd to have this aditional type (non-native like Date or Buffer) if I was the main user of this library I would like to have the type checking on a WebResource type, please let me know what do you think.

As for the 'write' typing above I am afraid I don't really understand what it does - My search seems that pine only uses the read types ([here](https://github.com/balena-io/pinejs/blob/9acd11e1ccfc7047c07e29258290d9a1f49b3018/src/bin/abstract-sql-compiler.ts#L25) and [here](https://github.com/balena-io-modules/abstract-sql-to-typescript/blob/c758673bd7a88923f1ce5f4f4aed2cf423d88f29/src/index.ts#L229))

Reply:
The write typing is what you can use for `.post`/`.patch` bodies vs the read typing which is what you get back from a `.get`. Usually they're the same but there can be some differences, the most notable being that reading a foreign key (object) is very different to writing it (number/etc). The difference could do with being more fleshed out but this module is only barely past being a PoC so the same could be said about most of it, it's still very handy though